src/auth/linking/consumer.test.ts

```diff
@@ -0,0 +1,442 @@
+
+import * as did from "../../../src/did/index.js"
+import * as consumer from "./consumer.js"
+import * as ucan from "../../ucan/index.js"
+import { LOCAL_IMPLEMENTATION } from "../local.js"
+import { setImplementations } from "../../setup.js"
+
+describe("generate temporary exchange key", async () => {
+  it("returns a temporary RSA key pair and DID", async () => {
+    const { temporaryRsaPair, temporaryDID } = await consumer.generateTemporaryExchangeKey()
+
+    expect(temporaryRsaPair).toBeDefined()
+    expect(temporaryRsaPair).not.toBeNull()
+    expect(temporaryDID).toBeDefined()
+    expect(temporaryDID).not.toBeNull()
+  })
+
+  it("returns a DID that matches the temporary RSA public key", async () => {
+    const { temporaryRsaPair, temporaryDID } = await consumer.generateTemporaryExchangeKey()
+    const temporaryPublicKey = await rsa.getPublicKey(temporaryRsaPair)
+    const didPublicKey = did.didToPublicKey(temporaryDID).publicKey
+
+    expect(didPublicKey).toEqual(temporaryPublicKey)
+  })
+})
+
+describe("handle session key", async () => {
+  let temporaryRsaPair: CryptoKeyPair
+  let temporaryDID: string
+  let sessionKey: CryptoKey
+  let exportedSessionKey: string
+  let encryptedSessionKey: ArrayBuffer
+  let iv: ArrayBuffer
+
+  beforeEach(async () => {
+    const cfg = config.normalize()
+    const { rsaSize, hashAlg } = cfg
+    temporaryRsaPair = await rsa.makeKeypair(rsaSize, hashAlg, KeyUse.Exchange)
+    sessionKey = await aes.makeKey({ alg: SymmAlg.AES_GCM, length: 256 })
+    exportedSessionKey = await aes.exportKey(sessionKey)
+    iv = utils.randomBuf(16)
+
+    const exportedPubKey = await rsa.getPublicKey(temporaryRsaPair)
+    temporaryDID = did.publicKeyToDid(exportedPubKey, did.KeyType.RSA)
+
+    const rawSessionKey = utils.arrBufToStr(utils.base64ToArrBuf(exportedSessionKey), CharSize.B16)
+    if (!temporaryRsaPair.publicKey) throw new Error("Temporary RSA public key missing")
+    encryptedSessionKey = await rsa.encrypt(rawSessionKey, temporaryRsaPair.publicKey)
+  })
+
+  it("returns a session key after validating a closed UCAN", async () => {
+    const closedUcan = await ucan.build({
+      issuer: await did.ucan(),
+      audience: temporaryDID,
+      lifetimeInSeconds: 60 * 5,
+      facts: [{ sessionKey: exportedSessionKey }],
+      potency: null
+    })
+    const msg = await aes.encrypt(ucan.encode(closedUcan), sessionKey, { iv, alg: SymmAlg.AES_GCM })
+    const message = JSON.stringify({
+      iv: utils.arrBufToBase64(iv),
+      msg,
+      sessionKey: utils.arrBufToBase64(encryptedSessionKey)
+    })
+    if (!temporaryRsaPair.privateKey) throw new Error("Temporary RSA private key missing")
+
+    const sessionKeyResult = await consumer.handleSessionKey(temporaryRsaPair.privateKey, message)
+
+    let val
+    if (sessionKeyResult.ok) { val = sessionKeyResult.value }
+
+    expect(sessionKeyResult.ok).toBe(true)
+    expect(val).toEqual(sessionKey)
```

Comment:
I usually go with this pattern:

```suggestion
    expect(sessionKeyResult.ok).toBe(true) // will already throw in case ok == false
    if (!sessionKeyResult.ok) throw "Invalid result" // but this is to make typescript happy
    
    expect(sessionKeyResult.value).toEqual(sessionKey)
```

But that's just to share my approach to it, I don't know which one is better :D

Reply:
Yeah, that is a nice approach because you don't need the `val` variable, with the tradeoff of the extra line for TypeScript. 🤔 

I'm not sure which approach is better either. Will keep these as is for now since they about equal to me.